lighthouse-core/test/report/v2/report-renderer-test.js

```diff
@@ -26,10 +26,55 @@ const sampleResults = require('../../results/sample_v2.json');
 const TEMPLATE_FILE = fs.readFileSync(__dirname + '/../../../report/v2/templates.html', 'utf8');
 
 describe('ReportRenderer V2', () => {
-  describe('renderReport', () => {
-    const document = jsdom.jsdom(TEMPLATE_FILE);
-    const renderer = new ReportRenderer(document);
+  const document = jsdom.jsdom(TEMPLATE_FILE);
+  const renderer = new ReportRenderer(document);
+
+  describe('createElement', () => {
+    it('creates a simple element using default values', () => {
+      const el = renderer._createElement('div');
+      assert.equal(el.localName, 'div');
+      assert.equal(el.className, '');
+      assert.equal(el.className, el.attributes.length);
+    });
+
+    it('creates an element from parameters', () => {
+      const el = renderer._createElement(
+          'div', 'class1 class2', {title: 'title attr', tabindex: 0});
+      assert.equal(el.localName, 'div');
+      assert.equal(el.className, 'class1 class2');
+      assert.equal(el.getAttribute('title'), 'title attr');
+      assert.equal(el.getAttribute('tabindex'), '0');
+    });
+  });
+
+  describe('cloneTemplate', () => {
+    it('should clone a template', () => {
+      const clone = renderer._cloneTemplate('#tmpl-lighthouse-audit-score');
+      assert.ok(clone.querySelector('.lighthouse-score'));
+    });
+
+    it('fails when template cannot be found', () => {
+      assert.throws(() => renderer._cloneTemplate('#unknown-selector'));
+    });
+  });
+
+  describe('_convertMarkdownLinksToElement', () => {
+    it('converts links', () => {
```

Comment:
A few more corner cases would be nice. With/without preamble text and text afterwards, etc